consensus/core/src/linearizer.rs

```diff
@@ -138,6 +144,49 @@ impl Linearizer {
         (sub_dag, commit)
     }
 
+    /// Calculates the commit's timestamp. If the median based timestamp calculation is enabled, then the timestamp will be calculated as the median of
+    /// leader's committed strong ancestors (leader.round - 1). Otherwise, the leader's timestamp will be used. To ensure that commit timestamp monotonicity is
+    /// respected it is compared against the `last_commit_timestamp_ms` and the maximum of the two is returned.
+    pub(crate) fn calculate_commit_timestamp(
+        context: &Context,
+        leader_block: &VerifiedBlock,
+        last_commit_timestamp_ms: BlockTimestampMs,
+        to_commit: &[VerifiedBlock],
+    ) -> BlockTimestampMs {
+        let timestamp_ms =
+            if context.protocol_config.consensus_median_based_timestamp() && to_commit.len() > 1 {
```

Comment:
I think the `to_commit.len() > 1` assumes the leader block is in to_commit. To make future changes easier, how about remove this condition and use `median(timestamps).unwrap_or(leader_block.timestamp_ms())`?